Public/FourKeyMetrics.ps1

```diff
@@ -149,49 +149,59 @@ function global:Get-ReleaseMetrics {
         [string[]]$subDirs,
         [Parameter(Mandatory=$true)]
         [string]$startDate,
-        [string[]]$ignoreReleases
+        [string[]]$ignoreReleases,
+        [string[]]$authors
     )
     $thisRelease = $releases[0]
     for ($i = 1; $i -lt $releases.Count; $i++) {
-        $lastRelease = $releases[$i]
+        $previousRelease = $releases[$i]
 
-        if (Assert-ReleaseShouldBeConsidered $ThisRelease.TagRef $ignoreReleases) {
-
-            $CommitAges = Get-CommitsBetweenTags $lastRelease.TagRef $thisRelease.TagRef $subDirs | Foreach-Object -Process { $thisRelease.Date - $_.Date } 
+        if (Assert-ReleaseNotIgnored $ThisRelease.TagRef $ignoreReleases) {
+            $CommitAges = Get-CommitsBetweenTags $previousRelease.TagRef $thisRelease.TagRef $subDirs $authors | Foreach-Object -Process { $thisRelease.Date - $_.Date } 
         }
         else {
-            
             $CommitAges = $null;
         }
-        
-        [PSCustomObject]@{
-                From             = $lastRelease.TagRef;
+
+        if ($null -eq $CommitAges){
+            Write-Warning "Release $($thisRelease.TagRef) has no relevant commits and will be ignored"
+        }
+        else {
+            [PSCustomObject]@{
+                From             = $previousRelease.TagRef;
                 To               = $thisRelease.TagRef;
-                FromDate         = $lastRelease.Date;
+                FromDate         = $previousRelease.Date;
                 ToDate           = $thisRelease.Date;
-                Interval         = $thisRelease.Date - $lastRelease.Date;
+                Interval         = $thisRelease.Date - $previousRelease.Date;
                 IsFix            = $thisRelease.IsFix;
                 CommitAges       = $CommitAges;
+            }
         }
 
-        if ($lastRelease.Date -le $startDate) {
+        if ($previousRelease.Date -le $startDate) {
             break
         }
 
-        $thisRelease = $lastRelease
+        $thisRelease = $previousRelease
     }
 }
 
-function Assert-ReleaseShouldBeConsidered($thisReleaseTagRef, $ignoreReleases) {
+function Assert-ReleaseNotIgnored($thisReleaseTagRef, $ignoreReleases) {
     return !($ignoreReleases | Where-Object {$thisReleaseTagRef -Like "refs/tags/$_"})
 }
 
 <#
 .SYNOPSIS
 Get a list of all commits added to master between two release tags
 #>
-function Get-CommitsBetweenTags($start, $end, $subDirs) {
-    $gitCommand = "git log --pretty=format:`"%h,%ai`" `"$start..$end`" --no-merges -- $subDirs"
+function Get-CommitsBetweenTags($start, $end, $subDirs, $authors) {
+    # Assume we're not filtering by authors, but build up a filter if we want one
+    $authorFilter = ""
+    foreach ($author in $authors) {
+        $authorFilter = $authorFilter + "--author=`"$author`" "
+    }
+
+    $gitCommand = "git log --pretty=format:`"%h,%ai`" `"$start..$end`" --no-merges $authorFilter -- $subDirs"
```

Comment:
Looks good. I guess there's a maximum length for this command line string, that Invoke-Expression can handle, but that would be a lot of authors before we have to run the line multiple times and append the results.

Reply:
Good point - hadn't considered that (at all!).

A quick google suggests the maximum length is either 8191 characters or 32764 characters, but I haven't looked into it 'properly'.

I'm happy to accept that and see if/when we hit an issue with the character limit.